lib/dep.ml

```diff
@@ -0,0 +1,43 @@
+(*
+ * Copyright (c) 2018 Ulysse Gérard <ulysse@gtarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+ open Compat
+
+type t = File of string | Dir of string
+
+let of_block block =
+  let open Block in
+  match directory block, file block, skip block with
+  | Some d, None, false -> Some (Dir d)
+  | None, Some f, false -> Some (File f)
+  | None, None, false -> None
+  | _, _, true -> None
+  | _ -> assert false
```

Comment:
The missing case is `Some d, Some f, false`. It can happen because these values are from labels that can be set together (and they work correctly).

Test case:

> \```ocaml dir=lib,file=block.mli
> \```

If you really need to return one value, the file should do. Note that it's relative to the value of `dir`.